python/lsst/fgcmcal/fgcmBuildStarsTable.py

```diff
@@ -0,0 +1,347 @@
+            tempCat['dec'][:] = df['decl'].values[use]
+            tempCat['x'][:] = df['x'].values[use]
+            tempCat['y'][:] = df['y'].values[use]
+            tempCat[visitKey][:] = df[self.config.visitDataRefname].values[use]
+            tempCat[ccdKey][:] = df[self.config.ccdDataRefName].values[use]
+
+            # Need to loop over ccds here
+            for detector in camera:
+                ccdId = detector.getId()
+                u = (tempCat[ccdKey] == ccdId)
+                tempCat['jacobian'][u] = approxPixelAreaFields[ccdId].evaluate(tempCat['x'][u],
+                                                                               tempCat['y'][u])
+                scaledInstFlux = (df[self.config.instFluxField].values[use[u]] *
+                                  visit['scaling'][ccdMapping[ccdId]])
+                tempCat[instMagKey][u] = (-2.5*np.log10(scaledInstFlux) + 2.5*np.log10(expTime))
+
+            # Compute instMagErr from instFluxErr/instFlux, any scaling
+            # will cancel out.
+
+            tempCat[instMagErrKey][:] = k*(df[self.config.instFluxField + 'Err'].values[use] /
+                                           df[self.config.instFluxField].values[use])
+
+            # Apply the jacobian if configured
+            if self.config.doApplyWcsJacobian:
+                tempCat[instMagKey][:] -= 2.5*np.log10(tempCat['jacobian'][:])
+
+            fullCatalog.extend(tempCat)
+
+            # Now do the aperture information
+            instMagIn = -2.5*np.log10(df[self.config.apertureInnerInstFluxField].values[use])
+            instMagErrIn = k*(df[self.config.apertureInnerInstFluxField + 'Err'].values /
+                              df[self.config.apertureInnerInstFluxField].values)
+            instMagOut = -2.5*np.log10(df[self.config.apertureOuterInstFluxField].values[use])
+            instMagErrOut = k*(df[self.config.apertureOuterInstFluxField + 'Err'].values /
+                               df[self.config.apertureOuterInstFluxField].values)
+            ok = (np.isfinite(instMagIn) & np.isfinite(instMagErrIn) &
+                  np.isfinite(instMagOut) & np.isfinite(instMagErrOut))
+
+            visit['deltaAper'] = np.median(instMagIn[ok] - instMagOut[ok])
+            visit['sources_read'] = 1
+
+            self.log.info("  Found %d good stars in visit %d (deltaAper = %0.3f)" %
+                          (use.size, visit['visit'], visit['deltaAper']))
+
+            if ((ctr % self.config.nVisitsPerCheckpoint) == 0 and
+               starObsDataRef is not None and visitCatDataRef is not None):
+                # We need to persist both the stars and the visit catalog which gets
+                # additional metadata from each visit.
+                starObsDataRef.put(fullCatalog)
+                visitCatDataRef.put(visitCat)
+
+        self.log.info("Found all good star observations in %.2f s" %
+                      (time.time() - startTime))
```

Comment:
I have found it very useful in the past, and it might be superfluous, but I don't want to retire it until we get through Gen3, since it's not harming anything.